aea/protocols/fipa/dialogues.py

```diff
@@ -201,16 +216,26 @@ def is_belonging_to_registered_dialogue(self, fipa_msg: Message, sender: Address
         :return: boolean indicating whether the message belongs to a registered dialogue
         """
         fipa_msg = cast(FIPAMessage, fipa_msg)
-        dialogue_id = cast(int, fipa_msg.get("dialogue_id"))
-        self_initiated_dialogue_label = DialogueLabel(dialogue_id, sender, agent_pbk)
-        other_initiated_dialogue_label = DialogueLabel(dialogue_id, sender, sender)
+        dialogue_reference = cast(Tuple[str, str], fipa_msg.get("dialogue_reference"))
```

Comment:
This method could be cleaned up eventually!

Reply:
Yes, for the moment maybe some comments in the docstring or the code can help